### PR TITLE
cmake: don't export grpc_cpp_plugin when crosscompiling

### PR DESCRIPTION
**Pull request by UVV-gh**
Plugin should not be exported for a cross-compilation environment,
because it has to run on a host/build system. If it's exported then
configure stage fails with an error, that imported target references
non-existing file. Even if the file is found, it can't be used on a host.

Signed-off-by: Vyacheslav Yurkov <uvv.mail@gmail.com>

@drfloob

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10844,11 +10844,14 @@ target_link_libraries(grpc_cpp_plugin
...
 
 if(gRPC_INSTALL)
-  install(TARGETS grpc_cpp_plugin EXPORT gRPCTargets
+  install(TARGETS grpc_cpp_plugin
     RUNTIME DESTINATION ${gRPC_INSTALL_BINDIR}
     LIBRARY DESTINATION ${gRPC_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${gRPC_INSTALL_LIBDIR}
   )
+  if(NOT CMAKE_CROSSCOMPILING)
+    install(TARGETS grpc_cpp_plugin EXPORT gRPCTargets)
```
Comment (jtattermusch) [inline CMakeLists.txt:10853]: what's the reasoning behind  not exporting grpc_cpp_plugin as gRPCTargets (as opposed to simply skipping the `install(TARGETS grpc_cpp_plugin ...)` entirely when crosscompiling)?

**Review by jtattermusch — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -10844,11 +10844,14 @@ target_link_libraries(grpc_cpp_plugin
 
 
 if(gRPC_INSTALL)
-  install(TARGETS grpc_cpp_plugin EXPORT gRPCTargets
+  install(TARGETS grpc_cpp_plugin
```
[the remark this answers is not in the document]
Comment (UVV-gh) [reply CMakeLists.txt:10847]: - Oh, I was not aware of a template. I take a look at it.
- I take a look at other plugins too. The idea of this PR was at least a starting point for the discussion ;)
- The other pitfall I see if someone is actually going to use a plugin on a target (not for cross-compilation). The a rollback would be required to locate the plugin manually as it's done now by gRPC itself

File: CMakeLists.txt (hunk 1 above)
Comment (UVV-gh) [reply CMakeLists.txt:10853]: We can do that too. You think it would be better for consistency between native and cross-compilation builds?

**Review by jtattermusch — APPROVED** [review]
LGTM once the tests are green.

**Comment by jtattermusch** [comment]
@veblush  please add another Googler's approval.

**Review by veblush — APPROVED** [review]

**Comment by xhochy** [comment]
With this change, you can no longer cross-compile gRPC and then use that cross-compiled gRPC on the target system as it is lacking the plugins. This is especially visible for us on conda-forge for Apple M1 users as there you can only cross-compile on CI (there is no native M1 CI yet) but all users use gRPC natively and not in a cross-compiling environment.

I opened an issue as a follow-up as this has been working fine with all previous gRPC versions: https://github.com/grpc/grpc/issues/26857

**Pull request merged**